### DISCUSSION TITLE
Gatsby develop gets stuck in an infinite loop, and also this error "Looks like develop for this site is already running, can you visit http://localhost:8000 ? If it is not, try again in five seconds!"

### DISCUSSION THREAD

**Discussion by lumenwrites** [gatsbyjs/gatsby#26869] [Help] [answered]
I'm getting the error:

```
Looks like develop for this site is already running, can you visit http://localhost:8000 ? If it is not, try again in five seconds!
```

I have tried killing all the node processes, deleting .cache/ and public/, updating and reinstalling all the node_packages, restarting my laptop - I'm still getting the same error. It's really weird. When I open http://localhost:8000, nothing is running there.

Can someone help me figure out what's wrong?

I have tried changing the port in package.json:

> "dev": "watch 'kill-port --port 8004 && gatsby develop  --port 8004'"

and things got even weirder - gatsby got stuck in the infinite loop. Printing out `Process on port 8004 killed` until I shut it down.

Something is already running at port 8000

```
⠦ source and transform nodes
/Users/ray/projects/mirage/node_modules/yoga-layout-prebuilt/yoga-layout/build/Release/nbind.js:53
        throw ex;
        ^

Error: write EPIPE
    at process.target._send (internal/child_process.js:835:20)
    at process.send (internal/child_process.js:706:19)
    at Timeout._onTimeout (/Users/ray/projects/mirage/node_modules/gatsby/src/commands/develop-process.ts:41:13)
    at listOnTimeout (internal/timers.js:551:17)
    at processTimers (internal/timers.js:494:7)
Emitted 'error' event on process instance at:
    at processEmit (/Users/ray/projects/mirage/node_modules/signal-exit/index.js:161:32)
    at process.emit (/Users/ray/projects/mirage/node_modules/@turist/fetch/dist/sourcemap-register.js:926:21)
    at internal/child_process.js:839:39
    at processTicksAndRejections (internal/process/task_queues.js:75:11)
    at runNextTicks (internal/process/task_queues.js:62:3)
    at processTimers (internal/timers.js:491:9) {
  errno: -32,
  code: 'EPIPE',
  syscall: 'write'
}
```

**Comment by SilencerWeb** [comment]
It was fixed in one of the latest versions. What version of gatsby are you running?

**Comment by madzadev** [comment]
@raymestalez getting the same error today on gatsby-cli 2.12.87 and gatsby 2.24.26, will check if updating helps 😉
edit: updated to the gatsby-cli 2.12.95 and gatsby 2.24.57, tho problem still persists ☹

**Comment by madzadev** [comment]
@raymestalez what worked for me after updating gatsby-cli and gatsby, was to initialize new test project with `gatsby new your-project-name` somewhere else (just to test if the new starter would work). Run `gatsby develop`, it should start the dev server, open it on http://localhost:8000. Now `cmd/ctrl+c` to close it and you can delete that test project.

Now go back to your project that didn't work and try `gatsby develop`, everything should work now.

**Comment by SilencerWeb** [comment]
@wardpeet could you look at this, please?

**Comment by sidharthachatterjee** [comment, marked as answer]
@raymestalez Hello! 

Apologies that you're seeing this. It appears that you might either have a process running or a stale lock file. Here is what we suggest:

- Delete the `~/.config/gatsby/sites` folder
- Run `ps aux | grep gatsby` to see any stale processes that you might have running tat are listening on port 8000
- Upgrade to latest gatsby

This was fixed but let us know if you're still seeing it!

**Reply by 0meram** [reply]
hi i would be happy to get some help with this issue , I tried what you said still not working/
 thankes

**Comment by sjnarmstrong** [comment]
I had the same issue and it seems to stop when i set:
```js
graphqlTypegen: false,
```
In the gatsby-config.ts so I assume that when the types are generated it gets detected by the autoreload step

**Reply by shareef-dweikat** [reply]
Worked for me! Thanks
